### PR TITLE
Add python extensions setting to workbench

### LINKED ISSUE #28253: Settings: More options
Settings Suggestions for Workbench:

- [x] Python Extensions (algorithms) from MantidPlot Settings > Mantid > Directories. This could go in Manage User Directories instead. https://forum.mantidproject.org/t/running-scripts-on-start-up-in-workbench/617
- [x] Drawstyle such as Steps
- [x] Colorbar Scale log/lin
- [x] Be able to set linestyle to None (currently could set linewidth to zero but this is a cheat!)
- [ ] Be able to hide the legend on plots
- [x] Set a default colormap

See here refs: #28500 
refs #27949 #27892

### PR DESCRIPTION
**Pull request by StephenSmith25**
**Description of work.**
This PR adds the python extensions setting to workbench. The new setting is located within the manage user directories widget. 

**To test:**
(1) Create a new directory (e.g mantid_scripts)
(2) Put a file (test.py) in there with the code:
`print("running \n running \n running \n running \n running \n running")`
(3) Set the python extensions directory to this new directory (manage user dirs -> python-> python extensions)
(4) Relaunch workbench
(5) The script should run on launch, evidenced by the printing of 'running' in the logger.

PARTLY Fixes #28253 . 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by alicerussell1 — CHANGES_REQUESTED** [review]
This looks good, works really well. Just one thing, when trying to add a second directory using the `Browse to Directory` button I would expect it to append the new directory to the old one with a semi-colon between rather than replace it as you might have one directory for algorithms and another for fit functions etc.

**Comment by StephenSmith25** [comment]
Might be wrong about this, but it seemed to me that the old setting only allowed one directory to be selected. So I wasn't sure if this was a technical limitation that when these scripts are executed its only expecting one directory to be stored in "user.python.plugins.directories". But then the setting is called ".directories" so it must allow more 🤔 . I'll have to find out, as yeah it would probably make sense to allow multiple directories.

**Comment by alicerussell1** [comment]
In the docs ([here](https://www.mantidproject.org/Loading_Extensions_On_Startup)) it says it is a semi-colon separated list of directories, so I assume it is meant to load multiple.

**Comment by StephenSmith25** [comment]
> 
> 
> In the docs ([here](https://www.mantidproject.org/Loading_Extensions_On_Startup)) it says it is a semi-colon separated list of directories, so I assume it is meant to load multiple.

Ah okay, thanks for looking that up - I shall get multiple selections implemented then.

**Comment by StephenSmith25** [comment]
Okay I've implemented multiple directory selections by using the list widget, used elsewhere in this settings widget. Had to rewrite more than I would have liked to in order to get it implemented. But should work now.

**Review by alicerussell1 — CHANGES_REQUESTED** [review]
Thanks for the changes, it works well. There's just a typo on the browse button.

File: qt/widgets/common/src/ManageUserDirectories.cpp [hunk 1]
```diff
@@ -111,6 +130,13 @@ void ManageUserDirectories::loadProperties() {
                             "defaultsave.directory"))
                         .trimmed();
   m_uiForm.leDefaultSave->setText(saveDir);
+
+  //// extensions directory
+  // QString extDir = QString::fromStdString(
+  //                     Mantid::Kernel::ConfigService::Instance().getString(
+  //                         "user.python.plugins.directories"))
+  //                     .trimmed();
+  // m_uiForm.leExtensions->setText(extDir);
```
[the remark this answers is not in the document]
Comment (StephenSmith25) [reply qt/widgets/common/src/ManageUserDirectories.cpp]: Oops forgot that was there 🤦‍♂️

**Review by alicerussell1 — APPROVED** [review]
Thanks for fixing those, this works well and code looks good to me 👍

**Pull request merged**